core-test/src/test/scala/org/bitcoins/core/psbt/PSBTSerializerTest.scala

```diff
@@ -220,6 +220,17 @@ class PSBTSerializerTest extends BitcoinSAsyncTest {
     succeed
   }
 
+  it must "fail to serialize PSBTs with unknown version numbers" in {
+    forAllAsync(PSBTGenerators.psbtWithVersion) { psbtF =>
```

Comment:
I think you should make a generator that specifically creates unknown versions for this test so that you don't need the if/else inside